src/main/java/roomescape/core/service/ReservationTimeService.java

```diff
@@ -0,0 +1,77 @@
+package roomescape.core.service;
+
+import static java.util.stream.Collectors.toSet;
+
+import java.util.List;
+import java.util.Set;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import roomescape.core.domain.Reservation;
+import roomescape.core.domain.ReservationTime;
+import roomescape.core.dto.ReservationTimeRequestDto;
+import roomescape.core.dto.ReservationTimeWithStateDto;
+import roomescape.core.repository.ReservationRepository;
+import roomescape.core.repository.ReservationTimeRepository;
+import roomescape.web.exception.BadRequestException;
+
+@Service
+public class ReservationTimeService {
+    private final ReservationTimeRepository reservationTimeRepository;
+    private final ReservationRepository reservationRepository;
+
+    public ReservationTimeService(
+            final ReservationTimeRepository reservationTimeRepository,
+            final ReservationRepository reservationRepository
+    ) {
+        this.reservationTimeRepository = reservationTimeRepository;
+        this.reservationRepository = reservationRepository;
+    }
+
+    @Transactional
+    public ReservationTime create(final ReservationTimeRequestDto request) {
+        final ReservationTime time = new ReservationTime(request.getStartAt());
+        validateDuplicatedStartAt(time);
+        final Long id = reservationTimeRepository.save(time);
+        return new ReservationTime(id, time.getStartAtString());
+    }
+
+    @Transactional(readOnly = true)
+    public List<ReservationTime> findAll() {
+        return reservationTimeRepository.findAll();
+    }
+
+    @Transactional(readOnly = true)
+    public List<ReservationTimeWithStateDto> findAllWithReservationState(final String date, final long themeId) {
+        List<ReservationTime> times = reservationTimeRepository.findAll();
+        Set<Long> reservedTimeIds = reservationRepository.findAllByDateAndThemeId(date, themeId)
+                .stream()
+                .map(Reservation::getTimeId)
+                .collect(toSet());
+        return times.stream()
+                .map(reservationTime -> toDto(reservationTime, reservedTimeIds))
+                .toList();
+    }
+
+    private ReservationTimeWithStateDto toDto(final ReservationTime time, final Set<Long> reservedTimeIds) {
+        if (reservedTimeIds.contains(time.getId())) {
+            return new ReservationTimeWithStateDto(time.getId(), time.getStartAtString(), true);
+        }
+        return new ReservationTimeWithStateDto(time.getId(), time.getStartAtString(), false);
+    }
```

Comment:
```suggestion
    private ReservationTimeWithStateDto toDto(final ReservationTime time, final Set<Long> reservedTimeIds) {
        boolean alreadyBooked = reservedTimeIds.contains(time.getId());
        return new ReservationTimeWithStateDto(time.getId(), time.getStartAtString(), alreadyBooked);
    }
```
변수로 빼는게 더 가독성 좋을것 같네요!